src/zenml/service_connectors/aws_service_connector.py

```diff
@@ -0,0 +1,1584 @@
+            # to get it from the caller identity
+            try:
+                session, _ = self._authenticate(self.auth_method)
+                sts_client = session.client("sts")
+                response = sts_client.get_caller_identity()
+            except ClientError as e:
+                raise AuthorizationException(
+                    "Failed to get ECR registry ID from ECR repository "
+                    f"name: {e}"
+                ) from e
+
+            registry_id = response["Account"]
+
+        return repository_name, registry_id
+
+    def _parse_eks_resource_id(self, resource_id: str) -> str:
+        """Validate and convert an EKS resource ID to an AWS region and EKS cluster name.
+
+        Args:
+            resource_id: The resource ID to convert.
+
+        Returns:
+            The EKS cluster name.
+
+        Raises:
+            ValueError: If the provided resource ID is not a valid EKS cluster
+                name or ARN.
+        """
+        # The resource ID could mean different things:
+        #
+        # - an EKS cluster ARN
+        # - an EKS cluster ID
+        #
+        # We need to extract the cluster name and region ID from the
+        # provided resource ID
+        config_region_id = self.config.region
+        cluster_name: Optional[str] = None
+        region_id: Optional[str] = None
+        if re.match(
+            r"^arn:aws:eks:[a-z0-9-]+:\d{12}:cluster/.+$",
+            resource_id,
+        ):
+            # The resource ID is an EKS cluster ARN
+            cluster_name = resource_id.split("/")[-1]
+            region_id = resource_id.split(":")[3]
+        elif re.match(
+            r"^[a-z0-9]+[a-z0-9_-]*$",
+            resource_id,
+        ):
+            # Assume the resource ID is an EKS cluster name
+            cluster_name = resource_id
+        else:
+            raise ValueError(
+                f"Invalid resource ID for a EKS cluster: {resource_id}. "
+                f"Supported formats are:\n"
+                f"EKS cluster ARN: arn:aws:eks:<region>:<account-id>:cluster/<cluster-name>\n"
+                f"ECR cluster name: <cluster-name>"
+            )
+
+        # If the connector is configured with a region and the resource ID
+        # is an EKS registry ARN or URI that specifies a different region
+        # we raise an error
+        if region_id and region_id != config_region_id:
+            raise ValueError(
+                f"The AWS region for the {resource_id} EKS cluster "
+                f"({region_id}) does not match the region configured in "
+                f"the connector ({config_region_id})."
+            )
+
+        return cluster_name
+
+    def _canonical_resource_id(
+        self, resource_type: str, resource_id: str
+    ) -> str:
+        """Convert a resource ID to its canonical form.
+
+        Args:
+            resource_type: The resource type to canonicalize.
+            resource_id: The resource ID to canonicalize.
```

Comment:
Haha that's definitely not a word...

Reply:
What's amazing is you making it all the way here down this gigantic file. Holy cow, this is huge.